### ISSUE TITLE
Stackoverflow on Gradle options tab

### ISSUE THREAD

**Issue by aplatypus** [apache/netbeans#3686]
### Apache NetBeans version

Apache NetBeans 12.6

### What happened

- Netbeans notificaton message popped-up with an error when I clicked APPLY on the Gradle options tab.
- Apparently the setting to use the projects gradlew as not/is not working.

**Expected ...**

1. Use the gradle project's `gradlew` when the "Prefer to use Gradle Wrapper that comes with the project"  is TICK-ed
2. Operate cleanly without a stackoverflow.

### How to reproduce

- Netbeans was not using my project's gradle wrapper (gradlew) to build
- I checked the Gradle tab, it has "Prefer to use Gradle Wrapper that comes with the project" => TICK
- I unticked that setting and then re-ticked the check-box.
- Pressed APPLY button
- A notificaton message popped-up with an error
- When I looked the error is a stackoverflow in Netbeans
>   ![image](https://user-images.githubusercontent.com/67028/156473130-e5b0836b-89b6-4606-9e68-86cddf277d9b.png)
- I have pasted in error information producted by Netbeans
- Incidently you need to fix the link in the error notification pop-ups to jump to Github

### Did this work correctly in an earlier version?

No

### Operating System

Ubuntu 20.04.4 LTS Release:	20.04

### JDK

Java: 17.0.1; building with Java 1.8

### Apache NetBeans packaging

Apache NetBeans binary zip

### Anything else

**Netbeans config** ...
``` 
Product Version: Apache NetBeans IDE 12.6
Java: 17.0.1; OpenJDK 64-Bit Server VM 17.0.1+12-LTS
Runtime: OpenJDK Runtime Environment 17.0.1+12-LTS
System: Linux version 5.13.0-30-generic running on amd64; UTF-8; en_AU (nb)
User directory: /home/william/.config/netbeans/v12.06/4ams/v17.00/user-02
Cache directory: /home/william/.cache/netbeans/v12.06/4ams/v17.00/cache-02
```

**Unexpected exception** pop-up ...
```
A java.lang.StackOverflowError exception has occurred.
Please report this at https://netbeans.apache.org/nb/report-issue,
including a copy of your messages.log file as an attachment.
The messages.log file is located in your /home/william/.config/netbeans/v12.06/4ams/v17.00/user-02/var/log folder.
```

I will attempt to attach logs as separate comment (if github permits it).

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by aplatypus** [comment]
**IDE log** ...
* The error is a stackoverflow, so tghe logis just looping
* I have truncated it after the pattern repeats.

```
INFO [org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl]: Load aiming FALLBACK for Gradle: :TmcsRelay[FALLBACK]
INFO [org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl]: Load aiming EVALUATED for Gradle: :TmcsRelay[FALLBACK]
INFO [null]: Last record repeated more than 10 times, further logs of this record are ignored until the log record changes.
INFO [null]: Last record repeated 6236 times in total.
SEVERE [org.openide.util.RequestProcessor]: Error in RequestProcessor org.netbeans.modules.gradle.NbGradleProjectImpl$1
java.lang.StackOverflowError
	at org.openide.loaders.FolderInstance.waitFinished(FolderInstance.java:307)
	at org.openide.loaders.FolderLookup$ProxyLkp.beforeLookup(FolderLookup.java:398)
	at org.openide.util.lookup.ProxyLookup.beforeLookup(ProxyLookup.java:268)
	at org.openide.util.lookup.ProxyLookup$R.myBeforeLookup(ProxyLookup.java:734)
	at org.openide.util.lookup.ProxyLookup$R.beforeLookup(ProxyLookup.java:756)
	at org.openide.util.lookup.ProxyLookup$R.myBeforeLookup(ProxyLookup.java:743)
	at org.openide.util.lookup.ProxyLookup$R.beforeLookup(ProxyLookup.java:756)
	at org.openide.util.lookup.ProxyLookup$LazyCollection.computeDelegate(ProxyLookup.java:1184)
	at org.openide.util.lookup.ProxyLookup$LazyCollection.access$900(ProxyLookup.java:1114)
	at org.openide.util.lookup.ProxyLookup$LazyCollection$1.hasNext(ProxyLookup.java:1314)
	at org.netbeans.spi.project.support.DelegatingLookupImpl.beforeLookup(DelegatingLookupImpl.java:92)
	at org.openide.util.lookup.ProxyLookup.lookup(ProxyLookup.java:273)
	at org.openide.util.lookup.SimpleProxyLookup.lookup(SimpleProxyLookup.java:134)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:388)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
            :
   at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.projectWithQuality(NbGradleProjectImpl.java:273)
	at org.netbeans.modules.gradle.NbGradleProjectImpl.getGradleProject(NbGradleProjectImpl.java:193)
	at org.netbeans.modules.gradle.loaders.AbstractProjectLoader$ReloadContext.<init>(AbstractProjectLoader.java:74)
	at org.netbeans.modules.gradle.loaders.GradleProjectLoaderImpl.loadProject(GradleProjectLoaderImpl.java:51)
[catch] at org.netbeans.modules.gradle.NbGradleProjectImpl.loadOwnProject0(NbGradleProjectImpl.java:396)
```

**Comment by aplatypus** [comment]
**UI log** ...

* attached .ZIP file:  [UI-log.xml.zip](https://github.com/apache/netbeans/files/8174167/UI-log.xml.zip)

**Comment by lkishalmi** [comment]
Thank you for reporting this.

The root cause has been fixed in #3480. It's time to upgrade to NetBeans 13!
